frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileWriter.java

```diff
@@ -0,0 +1,64 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.intellij.openapi.application.ReadAction;
+import com.intellij.openapi.vfs.VfsUtilCore;
+import com.intellij.openapi.vfs.VirtualFileManager;
+import io.vavr.collection.List;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import lombok.CustomLog;
+import lombok.val;
+
+import com.virtuslab.branchlayout.api.BranchLayout;
+import com.virtuslab.branchlayout.api.BranchLayoutException;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.branchlayout.impl.readwrite.BranchLayoutFileUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@CustomLog
+public class MacheteFileWriter {
+
+  @UIThreadUnsafe
```

Comment:
Given that this function is only supposed to be called within `WriteAction`, which are in turn executed on UI thread... shouldn't it rather be `@UIEffect`? and then `synchronized` won't be needed, as with all code that is only ever executed on UI thread (no risk of race conditions... that's why UI thread is designated in most UI frameworks in the first place)